### ISSUE TITLE
pycobertura install fails in fresh Debian buildpack-deps:stretch image

### ISSUE THREAD

**Issue by ericyhwang** [aconrad/pycobertura#114]
`buildpack-deps` is used as a base for many other images, like NodeJS's official images, and a pycobertura install fails in a fresh Debian Stretch (current LTS) image.

To run a temporary container with the image:

```
$ docker pull buildpack-deps:stretch
$ docker run --rm -it --entrypoint=bash buildpack-deps:stretch
```

Inside the container:

```
/# apt-get update && apt-get install python-pip -y
/# pip install pycobertura
Collecting pycobertura
  Downloading https://files.pythonhosted.org/packages/5d/4a/b993d4ac88ef16d20628ab1a28b4036205c54f178304c22d3900d5d072ad/pycobertura-2.0.1.tar.gz (63kB)
    100% |################################| 71kB 770kB/s 
    Complete output from command python setup.py egg_info:
    Couldn't find index page for 'setuptools_git' (maybe misspelled?)
    No local packages or working download links found for setuptools_git
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-OojLXp/pycobertura/setup.py", line 50, in <module>
        'pycobertura=pycobertura.cli:pycobertura'
      File "/usr/lib/python2.7/distutils/core.py", line 111, in setup
        _setup_distribution = dist = klass(attrs)
      File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 317, in __init__
        self.fetch_build_eggs(attrs['setup_requires'])
      File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 372, in fetch_build_eggs
        replace_conflicting=True,
      File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 846, in resolve
        dist = best[req.key] = env.best_match(req, ws, installer)
      File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 1118, in best_match
        return self.obtain(req, installer)
      File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 1130, in obtain
        return installer(requirement)
      File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 440, in fetch_build_egg
        return cmd.easy_install(req)
      File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", line 693, in easy_install
        raise DistutilsError(msg)
    distutils.errors.DistutilsError: Could not find suitable distribution for Requirement.parse('setuptools_git')
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-OojLXp/pycobertura/
```

The same install does succeed on `buildpack-deps:stable`, which is running Debian Buster.

Oddly, the same install command did succeed in the Debian Stretch image at around this time last year, so something must have broken between then and now. Even manually installing the older pycobertura from that time via `pip install pycobertura==0.10.5` fails in the same way.

**Comment by ericyhwang** [comment]
Alright, it looks like manually updating setuptools via `pip install -U setuptools` beforehand fixes the issue, so I'm going to close this since there is a workaround.

Still, it's odd that the install worked out-of-the-box last year, and is broken this year 🤔 

----

Here's the out-of-the-box setuptools versions for reference:

Debian Stretch `buildpack-deps:stretch` - broken:
```
/# python -c 'import setuptools; print(setuptools.__version__)'
33.1.1
```

Debian Buster `buildpack-deps:stable` - working:

```
/# python -c 'import setuptools; print(setuptools.__version__)'
40.8.0
```

**Comment by aconrad** [comment]
Hey @ericyhwang, thanks for bringing this up! I just ran a few tests myself (thanks for providing the docker commands, that was helpful) and came to the same conclusion as you did.

Indeed, I also found that the fix was to upgrade to a later version of `setuptools` is that _not_ provided by stretch.

I'm not sure what changed either since last year, but I did notice that the stretch image was updated 25 days ago (screenshot from the docker UI):

<img width="994" alt="Screen Shot 2021-06-05 at 10 15 15 AM" src="https://user-images.githubusercontent.com/247661/120899924-edefd580-c5e6-11eb-94db-b51305a377ca.png">

...  so maybe something changed there?

Possible workarounds:

1. Manually satisfy pycobertura's dependency:

```bash
apt-get update && apt-get install python-pip -y
pip install setuptools_git
pip install pycobertura
```

2. Upgrade setuptools to a newer version (cleanest IMO):

```bash
apt-get update && apt-get install python-pip -y
apt-get remove python-setuptools -y  # remove setuptools provided by the distro (optional step, but cleaner?)
pip install setuptools  # install latest setuptools from pypi (will shadow setuptools from the distro if previous step omitted)
pip install pycobertura
```

Otherwise, upgrade to Debian Buster since that seems to work. 😄
